cmd/upgrade.go

```diff
@@ -168,6 +170,18 @@ func (uc *upgradeCmd) loadCluster() error {
 		return errors.Wrap(err, "error parsing the api model")
 	}
 
+	// Use the Windows VHD associated with the aks-engine version if upgradeWindowsVHD is set to "true"
+	if uc.upgradeWindowsVHD {
```

Comment:
@marosset to review if this logic is what we want:

- if `--upgrade-windows-vhd=true`, then
  - set the appropriate Windows VHD defaults for this particular aks-engine build

Reply:
This same logic is *mostly* already duplicated twice

- https://github.com/Azure/aks-engine/blob/2569b075ebe7d7b54d93f2fc351b0290116ebb91/pkg/api/defaults.go#L840-L860
- https://github.com/Azure/aks-engine/blob/2569b075ebe7d7b54d93f2fc351b0290116ebb91/pkg/api/defaults.go#L895-L909

Can we maybe refactor that logic out to a reusable method?

I'm worried that if/when we update the logic it will get out of sync.

Reply:
My consideration for this is that defaults.go and upgrade.go handles different behaviors. One is to set the default value for the Windows VHD image reference, while the other one is to update/not update the Windows VHD image reference at a specific step. When we set the flag "upgrade-windows-vhd", it's not an indicator for default value. Does this make sense?

Reply:
Yes and I agree with you.